From 1ec30a9f97f03eef72aef7669bed12843901e9d7 Mon Sep 17 00:00:00 2001
From: Philipp Eder <philipp.eder@posteo.net>
Date: Sun, 27 Sep 2020 13:35:33 +0200
Subject: [PATCH] set font to fira code

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 6f05dce..51f54db 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Fira Code Retina:style=Retina:pixelsize=12";
 static int borderpx = 2;
 
 /*
-- 
2.26.2

