From 9146045ae742e391bc0b9296b66c524de07f8d3d Mon Sep 17 00:00:00 2001
From: Philipp Eder <philipp.eder@posteo.net>
Date: Sun, 11 Oct 2020 13:01:05 +0200
Subject: [PATCH] add method to resize inc, dec second master client

---
 config.def.h |  8 +++++---
 dwm.c        | 18 ++++++++++++++++++
 2 files changed, 23 insertions(+), 3 deletions(-)

diff --git a/config.def.h b/config.def.h
index e5b3966..bc58c4b 100644
--- a/config.def.h
+++ b/config.def.h
@@ -68,8 +68,10 @@ static Key keys[] = {
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
-	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
-	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
+	{ MODKEY,                       XK_h,      incclientmaster,  {.i = -5} },
+	{ MODKEY,                       XK_l,      incclientmaster,  {.i = +5} },
+	{ MODKEY|ShiftMask,             XK_h,      setmfact,       {.f = -0.05} },
+	{ MODKEY|ShiftMask,             XK_l,      setmfact,       {.f = +0.05} },
 	{ MODKEY,                       XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
 	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
diff --git a/dwm.c b/dwm.c
index 007e0be..f5e11b2 100644
--- a/dwm.c
+++ b/dwm.c
@@ -210,6 +210,7 @@ static void spawn(const Arg *arg);
 static void tag(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void vm_tile(Monitor *);
+static void incclientmaster(const Arg *arg);
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
@@ -1702,6 +1703,23 @@ vm_tile(Monitor *m)
 		}
 }
 
+void
+incclientmaster(const Arg *arg)
+{
+	unsigned int i, n;
+	Client *c;
+ 	for (n = 0, c = nexttiled(selmon->clients); c; c = nexttiled(c->next), n++);
+ 	if (n == 0)
+  		return;
+	int m = MIN(n, selmon->nmaster -1);
+	for (i = 0, c = nexttiled(selmon->clients); c && i < selmon->nmaster; c = nexttiled(c->next), i++){
+		if (i == m)
+			resizeclient(c, c->x + arg->i * m * -1, c->y, c->w + arg->i * m, c->h);
+		else
+			resizeclient(c, c->x, c->y, c->w + arg->i * -1, c->h);
+	}
+}
+
 
 void
 tile(Monitor *m)
-- 
2.26.2

